src/test/java/com/google/firebase/snippets/FirebaseAuthSnippets.java

```diff
@@ -657,14 +665,475 @@ public void generateSignInWithEmailLink() {
+      System.out.println("User: " + user.getUid());
+    }
+
+    // Iterating by pages 100 users at a time.
+    page = tenantAuth.listUsers(null, 100);
+    while (page != null) {
+      for (ExportedUserRecord user : page.getValues()) {
+        System.out.println("User: " + user.getUid());
+      }
+
+      page = page.getNextPage();
+    }
+    // [END list_all_users_tenant]
+  }
+
+  public void importWithHmacTenant(
+      TenantAwareFirebaseAuth tenantAuth) throws FirebaseAuthException {
+    // [START import_with_hmac_tenant]
+    List<ImportUserRecord> users = new ArrayList<>();
+    users.add(ImportUserRecord.builder()
+        .setUid("uid1")
+        .setEmail("user1@example.com")
+        .setPasswordHash("password-hash-1".getBytes())
+        .setPasswordSalt("salt1".getBytes())
+        .build());
+    users.add(ImportUserRecord.builder()
+        .setUid("uid2")
+        .setEmail("user2@example.com")
+        .setPasswordHash("password-hash-2".getBytes())
+        .setPasswordSalt("salt2".getBytes())
+        .build());
+    UserImportHash hmacSha256 = HmacSha256.builder()
+        .setKey("secret".getBytes())
+        .build();
+    UserImportResult result = tenantAuth.importUsers(users, UserImportOptions.withHash(hmacSha256));
+
+    for (ErrorInfo error : result.getErrors()) {
+      System.out.println("Failed to import user: " + error.getReason());
+    }
+    // [END import_with_hmac_tenant]
+  }
+
+  public void importWithoutPasswordTenant(
+      TenantAwareFirebaseAuth tenantAuth) throws FirebaseAuthException {
+    // [START import_without_password_tenant]
+    List<ImportUserRecord> users = new ArrayList<>();
+    users.add(ImportUserRecord.builder()
+        .setUid("some-uid")
+        .setDisplayName("John Doe")
+        .setEmail("johndoe@acme.com")
+        .setPhotoUrl("https://www.example.com/12345678/photo.png")
+        .setEmailVerified(true)
+        .setPhoneNumber("+11234567890")
+        // Set this user as admin.
+        .putCustomClaim("admin", true)
+        // User with SAML provider.
+        .addUserProvider(UserProvider.builder()
+            .setUid("saml-uid")
+            .setEmail("johndoe@acme.com")
+            .setDisplayName("John Doe")
+            .setPhotoUrl("https://www.example.com/12345678/photo.png")
+            .setProviderId("saml.acme")
+            .build())
+        .build());
+
+    UserImportResult result = tenantAuth.importUsers(users);
+
+    for (ErrorInfo error : result.getErrors()) {
+      System.out.println("Failed to import user: " + error.getReason());
+    }
+    // [END import_without_password_tenant]
+  }
+
+  public void verifyIdTokenTenant(TenantAwareFirebaseAuth tenantAuth, String idToken) {
+    // [START verify_id_token_tenant]
+    try {
+      // idToken comes from the client app
+      FirebaseToken token = tenantAuth.verifyIdToken(idToken);
+      // This should be set to TENANT-ID. Otherwise TenantIdMismatch error thrown.
```

Comment:
Should we clarify what we mean by `this` here?

Reply:
Done